doc/policy/mempool-design.md

```diff
@@ -0,0 +1,19 @@
+# Mempool design
+
+## Overview
+
+See
+https://delvingbitcoin.org/t/an-overview-of-the-cluster-mempool-proposal/393
+for a high level overview of the cluster mempool implementation (PR#33629,
+since v31.0) and its design rationale.
+
+## Feerate diagrams, replace by fee, and incentive compatibility
+
+See https://delvingbitcoin.org/t/mempool-incentive-compatibility/553 for an
+explanation of how we use feerate diagrams for mining, eviction, and evaluating
+transaction replacements.
+
+## Definitions and theorems
+
+See https://delvingbitcoin.org/t/cluster-mempool-definitions-theory/202 for
```

Comment:
In commit "doc: add design notes for cluster mempool"

I think the theory thread is outdated now. Merging and postlinearization won't be very relevant post SFL, and the existence of optimal linearizations can probably be shown much more concisely when using results from the GGT paper.

It's probably worth creating a trimmed-down version with just definitions of clusters, chunks, and fee-size diagams, possibly to include directly in the document here, but that can be done as a follow-up I think.

Reply:
Made an attempt at this in 9d25e2cbcd011b3046f9c3cd8de4e1b9112d0dfd